### PR TITLE
Fix FsStore::list and path parsing

### PR DESCRIPTION
**Pull request by teohhanhui**
(split out from #48, and more correct fix)

/cc @issackelly

### CODE REVIEW

File: src/fs_store.rs [hunk 1]
```diff
@@ -39,8 +40,9 @@ use error::ErrorKind;
 /// on other peers and for the splaying to be useful we need to guarantee a
 /// uniform distribution of documents across the subdirectories.
 ///
-/// Likewise we use the hex encoding of the document ID as the filename to avoid
-/// any issues with non-UTF8 characters in the document ID.
+/// Likewise we use the hex encoding of the document ID as the 2nd-level
+/// directory name to avoid any issues with non-UTF8 characters in the document
```
Comment (teohhanhui) [inline src/fs_store.rs:44]: This is actually a lie as the implementation doesn't support non-UTF8 characters at all (lol)

**Comment by teohhanhui** [comment]
Should I send separate PRs to appease clippy and cargo-deny? Or just add commits here? :smile:

File: src/fs_store.rs [hunk 2]
```diff
@@ -237,14 +239,14 @@ impl<'a> From<&'a DocumentId> for DocIdPaths {
 
 impl DocIdPaths {
     fn parse<P1: AsRef<Path>, P2: AsRef<Path>>(level1: P1, level2: P2) -> Option<Self> {
```
Comment (teohhanhui) [inline src/fs_store.rs:241]: Oh, there is indeed no reason for passing level1 and level2 paths separately. I'll change this.

**Comment by alexjg** [comment]
LGTM, once the linter is appeased in the other PR I'll merge this.

**Pull request merged**